### ISSUE TITLE
Define how SVG paint server units interact with CSS-inspired fill/stroke sizing properties

### ISSUE THREAD

**Issue by AmeliaBR** [w3c/svgwg#167]
E.g., if you have objectBoundingBox units versus userSpaceOnUse units, how are those resolved if you also have the new CSS background-image inspired properties in the fill/stroke property.

**Comment by tabatkins** [comment]
Are you talking about what's defined in <https://drafts.csswg.org/css-images-4/#paint-sources>?

**Comment by AmeliaBR** [comment]
@tabatkins Yep, pretty much. Didn't know that was in a spec.  

But now I'm concerned that the definition you have in CSS Images for `userSpaceOnUse` isn't consistent with how SVG gradients work.  It _is_ consistent with SVG patterns: the origin gets reset to the pattern tile in which you're painting, but the units don't get scaled.  But in gradients, you use the user space origin as well.

**Comment by AmeliaBR** [comment]
Given that SVG 2 no longer includes CSS images in paint & the sizing parameters that go with them, this is not a SVG 2 / CR issue; labels corrected accordingly.

It still needs to be resolved for the new fill / stroke paint spec (PS @tabatkins, have we got a stable link for that?  Not text decoration, something of its own.).  The same issues also come up in most of the FX specs that use SVG graphics effects elements: mask, clipPath, filter all need more clearly defined interactions between SVG sizing and CSS sizing.

**Comment by tabatkins** [comment]
> But now I'm concerned that the definition you have in CSS Images for userSpaceOnUse isn't consistent with how SVG gradients work. It is mostly consistent with SVG patterns: the origin gets reset to the pattern tile in which you're painting, but the units don't get scaled. But in gradients, you use the user space origin as well.

Right, those definitions are for paint servers used in CSS properties, where images intentionally don't control their own origin.  I guess it's not as useful to SVG's own properties, tho we need to resolve the contradiction for the fill/stroke properties that we want to share between SVG and CSS.

**Comment by tabatkins** [comment]
> It still needs to be resolved for the new fill / stroke paint spec (PS @tabatkins, have we got a stable link for that?  Not text decoration, something of its own.).

I don't understand what you mean by "not text decoration", but the current draft from fantasai and I (still experimental) is <https://drafts.csswg.org/css-text-decor-4/>.
